apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts

```diff
@@ -92,60 +135,74 @@ export class ExpandEmailEditorSchemaUsecase {
     return templateContent.length === 1 && templateContent[0].type === 'bulletList';
   }
 
-  private expendedForEach(node: TipTapNode & { attrs: { each: unknown } }): TipTapNode[] {
-    const eachObject = node.attrs.each;
-    const templateContent = node.content || [];
-
-    /*
-     * Due to maily limitations in the current implementation, the location of the for
-     * element is situated on the container of the list making the list a
-     *  child of the for element, if we iterate it we will get the
-     *  wrong behavior of multiple lists instead of list with multiple items.
-     * due to that when we choose the content to iterate in case we find a list we drill down additional level
-     * and iterate on the list items
-     * this prevents us from
-     * 1. item1
-     * 1. item2
-     *
-     * and turns it into
-     * 1.item1
-     * 2.item2
-     * which is the correct behavior
-     *
-     */
-    if ((this.isOrderedList(templateContent) || this.isBulletList(templateContent)) && templateContent[0].content) {
-      return [{ ...templateContent[0], content: this.regularExpansion(eachObject, templateContent[0].content) }];
-    }
+  /**
+   * For 'each' node, multiply the content by the number of items in the iterable array
+   * and add indexes to the placeholders.
+   *
+   * @example
+   * node:
+   * {
+   *   type: 'each',
+   *   attrs: { each: 'payload.comments' },
+   *   content: [
+   *     { type: 'variable', text: '{{ payload.comments.author }}' }
+   *   ]
+   * }
+   *
+   * variables:
+   * { payload: { comments: [{ author: 'John Doe' }, { author: 'Jane Doe' }] } }
+   *
+   * result:
+   * [
+   *   { type: 'text', text: '{{ payload.comments[0].author }}' },
+   *   { type: 'text', text: '{{ payload.comments[1].author }}' }
+   * ]
+   *
+   */
+  private multiplyForEachNode(
+    node: TipTapNode & { attrs: { each: unknown } },
+    variables: Record<string, unknown>
+  ): TipTapNode[] {
+    const iterablePath = node.attrs.each as string;
+    const nodeContent = node.content || [];
+    const expandedContent: TipTapNode[] = [];
 
-    return this.regularExpansion(eachObject, templateContent);
-  }
+    const iterableArray = this.getValueByPath(variables, iterablePath) as [{ [key: string]: unknown }];
```

Comment:
here we do not need to getValueByPath, id suggest doing the following:
go to [variableAttributeConfig](https://github.com/novuhq/novu/blob/nv-5034-NV-5138-fix-preview-of-maily-for-block/libs/application-generic/src/utils/process-node-attrs.ts#L42)
add a new entry, `{ attr: MailyAttrsEnum.EACH, flag: MailyAttrsEnum.EACH }`
the result will be that in the pipeline we will wrap each attribute (node.attrs.each) with `{{...}}` 
then you could get the value from the variable using liquid js.

```suggestion
    const iterableArray = await parseLiquid(iterablePath as string, variables);
```
let me know what you think.

Reply:
In this case, iterableArray accepts and returns this:
```ts
// variables, iterablePath = "payload.comments"
const iterableArray = this.getValueByPath(variables, iterablePath) as [{ [key: string]: unknown }];
/**
* iterableArray = [
*  {
*    name: '{{payload.comments.name}}',
* },
* {
*   name: '{{payload.comments.name}}',
* },
* {
*   name: '{{payload.comments.name}}',
* }
*]
**/
```

What you are suggesting, would return just the iterable path string, not the actual array:
```ts
const iterableArray = await parseLiquid(iterablePath as string, variables);
// iterableArray = payload.comments
```

Reply:
not sure if i understand it should return an array.

Editor:
``` 
for block {{ payload.comments }}
```

Variable:
```
{"payload" : {
    "comments": [
      " a_1 ",
      " a_2 ",
      " a_3 "
    ]
  }
}
```

![image](https://github.com/user-attachments/assets/815a1b8e-6d6a-4105-9184-120e71b92c35)

Reply:
It currently doesnt work like this because we dont have `for block {{ payload.comments }}` anywhere in the maily json, we have just `each: {{ payload.comments}}`. That said its interesting idea to explore I will have a look if we can do this separately.

Reply:
for semantics representation aside, this one should work, don't it?
we have `each: {{ payload.comments}}`
with data variables: `{"payload":{"comments":[" a_1 "," a_2 "," a_3 "]}}`
once we execute `const iterableArray = await parseLiquid(iterablePath as string, variables);`
we should get the array `[" a_1 "," a_2 "," a_3 "]`